File: browser-mcp-extension/icons/generate_icons.py
#!/usr/bin/env python3
"""
Browser MCP Icon Generator
Creates PNG icons using PIL (Python Imaging Library)
"""

try:
    from PIL import Image, ImageDraw, ImageFont
    import os
except ImportError:
    print("❌ PIL not installed. Install with: pip3 install pillow")
    print("\nOr use the HTML generator: open create-icons.html")
    exit(1)

def create_icon(size):
    """Create an icon of the specified size"""
    
    # Create image with gradient background (approximate with solid color)
    img = Image.new('RGBA', (size, size), (102, 126, 234, 255))
    draw = ImageDraw.Draw(img)
    
    # Draw gradient effect (top to bottom)
    for y in range(size):
        ratio = y / size
        r = int(102 + (118 - 102) * ratio)
        g = int(126 + (75 - 126) * ratio)
        b = int(234 + (162 - 234) * ratio)
        draw.line([(0, y), (size, y)], fill=(r, g, b, 255))
    
    # Draw rounded rectangle mask
    mask = Image.new('L', (size, size), 0)
    mask_draw = ImageDraw.Draw(mask)
    corner_radius = int(size * 0.2)
    mask_draw.rounded_rectangle(
        [(0, 0), (size, size)],
        corner_radius,
        fill=255
    )
    
    # Apply mask
    img.putalpha(mask)
    
    # Draw "M" text
    try:
        # Try to use a bold font
        font_size = int(size * 0.6)
        try:
            font = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", font_size)
        except:
            font = ImageFont.load_default()
    except:
        font = ImageFont.load_default()
    
    text = "M"
    bbox = draw.textbbox((0, 0), text, font=font)
    text_width = bbox[2] - bbox[0]
    text_height = bbox[3] - bbox[1]
    text_x = (size - text_width) // 2
    text_y = (size - text_height) // 2 - bbox[1]
    
    draw.text((text_x, text_y), text, fill=(255, 255, 255, 255), font=font)
    
    # Add badge for larger sizes
    if size >= 48:
        badge_size = int(size * 0.3)
        badge_x = int(size * 0.75)
        badge_y = int(size * 0.25)
        
        # Draw green circle
        draw.ellipse(
            [(badge_x - badge_size//2, badge_y - badge_size//2),
             (badge_x + badge_size//2, badge_y + badge_size//2)],
            fill=(76, 175, 80, 255)
        )
        
        # Draw "33" text
        badge_font_size = int(size * 0.12)
        try:
            badge_font = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", badge_font_size)
        except:
            badge_font = ImageFont.load_default()
        
        badge_text = "33"
        badge_bbox = draw.textbbox((0, 0), badge_text, font=badge_font)
        badge_text_width = badge_bbox[2] - badge_bbox[0]
        badge_text_height = badge_bbox[3] - badge_bbox[1]
        badge_text_x = badge_x - (badge_text_width // 2 if badge_text_width > 0 else 0)
        badge_text_y = badge_y - (badge_text_height // 2 if badge_text_height > 0 else 0) - badge_bbox[1]
        
        draw.text((badge_text_x, badge_text_y), badge_text, 
                 fill=(255, 255, 255, 255), font=badge_font)
    
    return img

def main():
    print("🎨 Browser MCP Icon Generator\n")
    
    sizes = [16, 48, 128]
    script_dir = os.path.dirname(os.path.abspath(__file__))
    
    for size in sizes:
        try:
            img = create_icon(size)
            filename = os.path.join(script_dir, f'icon-{size}.png')
            img.save(filename, 'PNG')
            print(f"✅ Created: icon-{size}.png")
        except Exception as e:
            print(f"❌ Error creating icon-{size}.png: {e}")
    
    print("\n🎉 All icons generated successfully!")
    print(f"Icons saved to: {script_dir}")
    print("\nNext steps:")
    print("1. Reload your extension in chrome://extensions/")
    print("2. Icons should now appear correctly!")

if __name__ == "__main__":
    main()

File: browser-mcp-extension/icons/generate-icons.js
#!/usr/bin/env node

/**
 * Generate Browser MCP Icons
 * Creates PNG icons using node-canvas (if available) or provides instructions
 */

const fs = require('fs');
const path = require('path');

console.log('🎨 Browser MCP Icon Generator\n');

// Check if canvas is available
let Canvas;
try {
    Canvas = require('canvas');
} catch (e) {
    console.log('📦 node-canvas not installed. Using alternative method...\n');
    generateFallbackIcons();
    process.exit(0);
}

function drawIcon(size) {
    const canvas = Canvas.createCanvas(size, size);
    const ctx = canvas.getContext('2d');
    
    // Background - gradient effect (approximate with solid color)
    const gradient = ctx.createLinearGradient(0, 0, size, size);
    gradient.addColorStop(0, '#667eea');
    gradient.addColorStop(1, '#764ba2');
    ctx.fillStyle = gradient;
    ctx.fillRect(0, 0, size, size);
    
    // Round corners
    ctx.globalCompositeOperation = 'destination-in';
    const cornerRadius = size * 0.2;
    ctx.beginPath();
    ctx.moveTo(cornerRadius, 0);
    ctx.lineTo(size - cornerRadius, 0);
    ctx.quadraticCurveTo(size, 0, size, cornerRadius);
    ctx.lineTo(size, size - cornerRadius);
    ctx.quadraticCurveTo(size, size, size - cornerRadius, size);
    ctx.lineTo(cornerRadius, size);
    ctx.quadraticCurveTo(0, size, 0, size - cornerRadius);
    ctx.lineTo(0, cornerRadius);
    ctx.quadraticCurveTo(0, 0, cornerRadius, 0);
    ctx.closePath();
    ctx.fill();
    
    ctx.globalCompositeOperation = 'source-over';
    
    // Draw "M" letter
    ctx.fillStyle = '#ffffff';
    ctx.font = `bold ${size * 0.6}px Arial`;
    ctx.textAlign = 'center';
    ctx.textBaseline = 'middle';
    ctx.fillText('M', size / 2, size / 2);
    
    // Add badge for larger sizes
    if (size >= 48) {
        const badgeSize = size * 0.3;
        const badgeX = size * 0.75;
        const badgeY = size * 0.25;
        
        ctx.beginPath();
        ctx.arc(badgeX, badgeY, badgeSize / 2, 0, 2 * Math.PI);
        ctx.fillStyle = '#4CAF50';
        ctx.fill();
        
        ctx.fillStyle = '#ffffff';
        ctx.font = `bold ${size * 0.12}px Arial`;
        ctx.fillText('33', badgeX, badgeY);
    }
    
    return canvas;
}

function generateFallbackIcons() {
    console.log('Alternative Methods to Create Icons:\n');
    
    console.log('Option 1: Use the HTML Generator');
    console.log('  1. Open: icons/create-icons.html in your browser');
    console.log('  2. Click "Download All Icons"');
    console.log('  3. Save files to the icons/ folder\n');
    
    console.log('Option 2: Use ImageMagick (if installed)');
    console.log('  brew install imagemagick');
    console.log('  Then run:');
    console.log('    convert -size 16x16 xc:"#667eea" -fill white -font Arial-Bold');
    console.log('    -pointsize 10 -gravity center -annotate +0+0 "M" icon-16.png\n');
    
    console.log('Option 3: Use online tool');
    console.log('  - Visit: https://www.favicon-generator.org/');
    console.log('  - Upload any image with "M" text');
    console.log('  - Download 16x16, 48x48, 128x128 sizes\n');
    
    console.log('Option 4: Simple colored squares (temporary)');
    console.log('  Creating simple placeholder icons...\n');
    
    createSimplePlaceholders();
}

function createSimplePlaceholders() {
    // Create simple SVG placeholders that Chrome can use
    const sizes = [16, 48, 128];
    
    sizes.forEach(size => {
        const svg = `
<svg width="${size}" height="${size}" xmlns="http://www.w3.org/2000/svg">
  <defs>
    <linearGradient id="grad" x1="0%" y1="0%" x2="100%" y2="100%">
      <stop offset="0%" style="stop-color:#667eea;stop-opacity:1" />
      <stop offset="100%" style="stop-color:#764ba2;stop-opacity:1" />
    </linearGradient>
  </defs>
  <rect width="${size}" height="${size}" rx="${size * 0.2}" fill="url(#grad)"/>
  <text x="50%" y="50%" font-family="Arial" font-size="${size * 0.6}" font-weight="bold" 
        fill="white" text-anchor="middle" dominant-baseline="central">M</text>
  ${size >= 48 ? `
  <circle cx="${size * 0.75}" cy="${size * 0.25}" r="${size * 0.15}" fill="#4CAF50"/>
  <text x="${size * 0.75}" y="${size * 0.25}" font-family="Arial" font-size="${size * 0.12}" 
        font-weight="bold" fill="white" text-anchor="middle" dominant-baseline="central">33</text>
  ` : ''}
</svg>`;
        
        const filename = path.join(__dirname, `icon-${size}.svg`);
        fs.writeFileSync(filename, svg.trim());
        console.log(`✅ Created: icon-${size}.svg`);
    });
    
    console.log('\n⚠️  Note: Chrome extensions need PNG files.');
    console.log('Convert SVG to PNG using:');
    console.log('  - Online: https://svgtopng.com/');
    console.log('  - macOS: Open SVG in Preview, Export as PNG');
    console.log('  - Command: brew install librsvg && rsvg-convert icon.svg > icon.png\n');
}

// Try to generate PNG icons
try {
    const sizes = [16, 48, 128];
    
    sizes.forEach(size => {
        const canvas = drawIcon(size);
        const buffer = canvas.toBuffer('image/png');
        const filename = path.join(__dirname, `icon-${size}.png`);
        fs.writeFileSync(filename, buffer);
        console.log(`✅ Created: icon-${size}.png`);
    });
    
    console.log('\n🎉 All icons generated successfully!');
    console.log('Icons saved to:', __dirname);
    
} catch (error) {
    console.error('Error generating PNG icons:', error.message);
    console.log('\nFalling back to alternative methods...\n');
    generateFallbackIcons();
}

